### ISSUE TITLE
Unable to run resolveBuildDependencies in parallel

### ISSUE THREAD

**Issue by anuraaga** [gogradle/gogradle#162]
In a project with multiple gogradle subprojects and parallel build is enabled, resolveBuildDependencies failed on a file lock exception. Ideally they can use separate locks, otherwise, the second task should wait for the lock instead of failing.

```
Caused by: java.nio.channels.OverlappingFileLockException
        at com.github.blindpirate.gogradle.core.cache.DefaultGlobalCacheManager.doGetMetadata(DefaultGlobalCacheManager.java:119)
        at com.github.blindpirate.gogradle.core.cache.DefaultGlobalCacheManager.getMetadata(DefaultGlobalCacheManager.java:106)
        at com.github.blindpirate.gogradle.core.pack.GlobalCachePackagePathResolver.produce(GlobalCachePackagePathResolver.java:51)
        at com.github.blindpirate.gogradle.util.logging.DebugLogMethodInterceptor.invoke(DebugLogMethodInterceptor.java:38)
        at com.github.blindpirate.gogradle.core.pack.GlobalCachePackagePathResolver.produce(GlobalCachePackagePathResolver.java:35)
        at com.github.blindpirate.gogradle.util.FactoryUtil.lambda$produce$0(FactoryUtil.java:29)
        at com.github.blindpirate.gogradle.util.FactoryUtil.produce(FactoryUtil.java:32)
        at com.github.blindpirate.gogradle.core.pack.DefaultPackagePathResolver.produce(DefaultPackagePathResolver.java:61)
        at com.github.blindpirate.gogradle.util.logging.DebugLogMethodInterceptor.invoke(DebugLogMethodInterceptor.java:38)
        at com.github.blindpirate.gogradle.core.dependency.parse.DefaultMapNotationParser.parse(DefaultMapNotationParser.java:67)
        at com.github.blindpirate.gogradle.util.DependencySetUtils.parseMany(DependencySetUtils.java:30)
        at com.github.blindpirate.gogradle.core.dependency.produce.ExternalDependencyFactory.produce(ExternalDependencyFactory.java:65)
        at com.github.blindpirate.gogradle.core.dependency.produce.strategy.GogradleRootProduceStrategy.determineDependencies(GogradleRootProduceStrategy.java:78)
        at com.github.blindpirate.gogradle.core.dependency.produce.strategy.GogradleRootProduceStrategy.produce(GogradleRootProduceStrategy.java:62)
        at com.github.blindpirate.gogradle.util.logging.DebugLogMethodInterceptor.invoke(DebugLogMethodInterceptor.java:38)
        at com.github.blindpirate.gogradle.task.ResolveDependencies.produceFirstLevelDependencies(ResolveDependencies.java:175)
        at com.github.blindpirate.gogradle.task.ResolveDependencies.resolveDependencies(ResolveDependencies.java:169)
        at com.github.blindpirate.gogradle.task.ResolveDependencies.resolve(ResolveDependencies.java:152)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:73)
        at org.gradle.api.internal.project.taskfactory.StandardTaskAction.doExecute(StandardTaskAction.java:46)
        at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:39)
        at org.gradle.api.internal.project.taskfactory.StandardTaskAction.execute(StandardTaskAction.java:26)

```

**Comment by blindpirate** [comment]
Thanks for reporting. That's a corner case I missed. It would be fixed in December release.

**Comment by blindpirate** [comment]
Fixed by `0.8.0`. Any feedbacks are welcomed.
